src/cript/api/api.py

```diff
@@ -511,22 +510,42 @@ def save(self, project: Project) -> None:
 
         Returns
         -------
-        None
+        A set of extra saved node UUIDs.
             Just sends a `POST` or `Patch` request to the API
         """
+        if known_uuid is None:
+            known_uuid = set()
 
-        project.validate()
-
+        node.validate()
         # saves all the local files to cloud storage right before saving the Project node
         # Ensure that all file nodes have uploaded there payload before actual save.
-        for file_node in project.find_children({"node": ["File"]}):
+        for file_node in node.find_children({"node": ["File"]}):
             file_node.ensure_uploaded(api=self)
 
-        response: Dict = requests.post(url=f"{self._host}/{project.node_type.lower()}", headers=self._http_headers, data=project.json).json()
+        node_known = len(self.search(type(node), SearchModes.UUID, str(node.uuid)).current_page_results) == 1
 
+        json_data = node.get_json(known_uuid=known_uuid).json
+        if node_known:
+            response: Dict = requests.patch(url=f"{self._host}/{node.node_type.lower()}", headers=self._http_headers, data=json_data).json()
```

Comment:
I am guessing we'll have to send it in a specific order because otherwise we can send the project as patch and it would give us issues

Reply:
No, that shouldn't be necessary.
The only thing that I haven't handled yet, are `duplicate node` errors.
Everything else should automatically put itself in the right order.

Reply:
okie dokes, that sounds good. It feels too simple though